diff --git a/cedar-lean/Cedar/Spec/Ext/Decimal.lean b/cedar-lean/Cedar/Spec/Ext/Decimal.lean
index 949d72d..1722155 100644
--- a/cedar-lean/Cedar/Spec/Ext/Decimal.lean
+++ b/cedar-lean/Cedar/Spec/Ext/Decimal.lean
@@ -56,7 +56,7 @@ public def parse (str : String) : Option Decimal :=
       | .some l, .some r =>
         let l' := l * (Int.pow 10 DECIMAL_DIGITS)
         let r' := r * (Int.pow 10 (DECIMAL_DIGITS - rlen))
-        let i  := if !left.startsWith "-" then l' + r' else l' - r'
+        let i  := if l ≥ 0 then l' + r' else l' - r'
         decimal? i
       | _, _ => .none
     else .none
