diff --git a/cedar-lean/Cedar/SymCC/Encoder.lean b/cedar-lean/Cedar/SymCC/Encoder.lean
index fa67176..8259e6a 100644
--- a/cedar-lean/Cedar/SymCC/Encoder.lean
+++ b/cedar-lean/Cedar/SymCC/Encoder.lean
@@ -266,10 +266,7 @@ def encodePattern : Pattern → EncoderM String
 
 public def defineEntity (tyEnc : String) (entity : EntityUID) : EncoderM String := do
   match (← get).enums.find? entity.ty with
-  | .some members =>
-    match members.idxOf? entity.eid with
-    | .some enum_idx => return s!"{enumId tyEnc enum_idx}"
-    | .none          => throw (IO.userError s!"Unknown enum member `{entity.eid}`, expected one of {members}")
+  | .some members => return s!"{enumId tyEnc (members.idxOf entity.eid)}"
   | .none         => defineTerm tyEnc s!"({tyEnc} \"{← encodeString entity.eid}\")"
 
 private def indexOfAttr (a : Attr) : TermType → EncoderM Nat
