diff --git a/cedar-lean/Cedar/SymCC/Encoder.lean b/cedar-lean/Cedar/SymCC/Encoder.lean
--- a/cedar-lean/Cedar/SymCC/Encoder.lean
+++ b/cedar-lean/Cedar/SymCC/Encoder.lean
@@ -227,7 +227,7 @@ def defineSet (tyEnc : String) (tEncs : List String) : EncoderM String := do
     defineTerm tyEnc s!"(set.insert {tEncs} (as set.empty {tyEnc}))"

 public def defineRecord (tyEnc : String) (tEncs : List String) : EncoderM String := do
-  defineTerm tyEnc (if tEncs.isEmpty then tyEnc else s!"({tyEnc} {String.intercalate " " tEncs})")
+  defineTerm tyEnc s!"({tyEnc} {String.intercalate " " tEncs})"

 def encodeUUF (uuf : UUF) : EncoderM String := do
   if let (.some enc) := (← get).uufs.find? uuf then return enc
