diff --git a/cedar-lean/Cedar/Validation/Levels.lean b/cedar-lean/Cedar/Validation/Levels.lean
index 3b03e3c..5647ae7 100644
--- a/cedar-lean/Cedar/Validation/Levels.lean
+++ b/cedar-lean/Cedar/Validation/Levels.lean
@@ -58,7 +58,7 @@ public def TypedExpr.checkEntityAccessLevel (tx : TypedExpr) (env : TypeEnv) (n
   match tx, path with
   | .var _ _, _ => true
   | .lit (.entityUID euid) _, _ =>
-    euid == env.reqty.action
+    false
   | .ite tx₁ tx₂ tx₃ _, _ =>
     tx₁.checkLevel env nmax &&
     tx₂.checkEntityAccessLevel env n nmax path &&
